chronos/asyncloop.nim

```diff
@@ -12,13 +12,20 @@ include "system/inclrtl"
 
 import os, tables, strutils, heapqueue, lists, options, nativesockets, net,
        deques
-import metrics
+when defined(metrics):
+  import metrics
+  when defined(threads):
+    import locks
 import ./timer, ./srcloc
 
 export Port, SocketFlag
 export timer
 
-var callbacksByFuture* = initCountTable[string]()
+when defined(metrics):
+  var callbacksByFuture* = initCountTable[string]()
+  when defined(threads):
+    var callbacksByFutureLock*: Lock
```

Comment:
`callbacksByFuture` must not be global, because async dispatcher will be single-threaded even in multi-threaded environment, so for N threads there will be present N dispatchers.

Reply:
Also because dispatcher will be single-threaded locks are not needed.

Reply:
> Also because dispatcher will be single-threaded locks are not needed.

You can have multiple event loops running in different threads and collect metrics from all of them.